lib/thread.c

```diff
@@ -786,6 +788,12 @@ struct thread *funcname_thread_add_read_write(int dir, struct thread_master *m,
 
 		thread = thread_get(m, dir, func, arg, debugargpass);
 
+		/* If this is new FD then clean up the index before reuse */
```

Comment:
Why not put this on the memmove?  We know we should clean up the last spot right then, right?

Reply:
We can do it at two places. One when we reuse it or when we do memmove. I thought reset before using it. I wan change it to memmove.